### PR TITLE
Documentation: Data Retention FAQ

### PR DESCRIPTION
**Pull request by fakela**
#### Summary

Documentation: Data Retention FAQ 

#### Ticket Link

  Fixes https://github.com/mattermost/docs/issues/3647

### CODE REVIEW

File: source/administration/data-retention.rst [hunk 1]
```diff
@@ -108,3 +108,19 @@ Does the data retention job include archived channels?
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Posts and attachments in archived channels are affected by the data retention job. If a post exceeds the age configured for the data retention job it will be deleted from the database.
+
+How long does it take to run a deletion query and does it affect server performance?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Data retention runs the actual deletion query in batches, deleting data in blocks of 1,000 records per query so the database won’t be locked up for extended periods of time with slow, long-running queries (keeping to this limit keeps the query down to a few milliseconds execution time on the database itself).
```
Comment (justinegeffen) [inline source/administration/data-retention.rst:115]:
```suggestion
Data retention runs the actual deletion query in batches, deleting data in blocks of 1 000 records per query. This is so the database won’t be locked up for extended periods of time with long-running queries. Keeping to this limit keeps the query down to a few milliseconds' execution time on the database itself.
```

File: source/administration/data-retention.rst [hunk 2]
```diff
@@ -108,3 +108,19 @@ Does the data retention job include archived channels?
...
+
+Each batch of data is deleted based on indexes, so the queries are generally quite quick to execute on such small batches meaning the server will stay fully responsive while the process is running.
```
Comment (justinegeffen) [inline source/administration/data-retention.rst:117]:
```suggestion
Each batch of data is deleted based on indexes, meaning the queries are generally quick to execute on such small batches. This helps the server remain fully responsive while the process is running.
```

File: source/administration/data-retention.rst [hunk 3]
```diff
@@ -108,3 +108,19 @@ Does the data retention job include archived channels?
...
+
+How do I know whether the data retention job is running/scheduled?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When the job scheduler identifies that it is time for the data retention job to be scheduled again for immediate running (i.e. at the time they have configured for it to run each day), a debug level logline is printed: ``Scheduling data retention job``.
```
Comment (justinegeffen) [inline source/administration/data-retention.rst:122]:
```suggestion
The job scheduler runs the data retention job based on the time specified in the configuration settings. At this time a ``DEBUG``-level log line is printed: ``Scheduling data retention job``.
```

File: source/administration/data-retention.rst [hunk 4]
```diff
@@ -108,3 +108,19 @@ Does the data retention job include archived channels?
...
+
+When a job server then picks up that scheduled job in order to execute it there will be a debug log as follows: ``Worker EnterpriseDataRetention: Received a new candidate job``.
```
Comment (justinegeffen) [inline source/administration/data-retention.rst:124]:
```suggestion
When a job server picks up that scheduled job in order to execute it, a ``DEBUG``-level log line is generated: ``Worker EnterpriseDataRetention: Received a new candidate job``.
```

File: source/administration/data-retention.rst [hunk 5]
```diff
@@ -108,3 +108,19 @@ Does the data retention job include archived channels?
...
+
+When the job is complete there is an info level logline: ``Worker EnterpriseDataRetention: Job is complete``.
```
Comment (justinegeffen) [inline source/administration/data-retention.rst:126]:
```suggestion
When the job is complete, an ``INFO``-level log line ``Worker EnterpriseDataRetention: Job is complete`` is generated. 
```

File: source/administration/data-retention.rst (hunk 1 above)
Comment (fakela) [reply source/administration/data-retention.rst:115]: This looks good

File: source/administration/data-retention.rst (hunk 2 above)
Comment (fakela) [reply source/administration/data-retention.rst:117]: Each batch of data is deleted based on indexes making the queries quick to execute on small batches. This helps the server remain fully responsive while the process is running

@justinegeffen here is my suggestion what do you think?

File: source/administration/data-retention.rst (hunk 4 above)
Comment (fakela) [reply source/administration/data-retention.rst:124]: When a job server picks up that scheduled job for execution, a ``DEBUG`` -level log line is generated: ``Worker EnterpriseDataRetention: Received a new candidate job``.

@justinegeffen my second suggestion

**Comment by fakela** [comment]
> Thanks, @fakela! I've made some rewording suggestions which I'd appreciate your feedback on, as I want to make sure I haven't affected any of the terminology or steps with my changes. :)

@justinegeffen thanks a lot for the suggestions, it was helpful. I dropped a suggestion on some looking forward to your feedback

File: source/administration/data-retention.rst [hunk 6]
```diff
@@ -108,3 +108,19 @@ Does the data retention job include archived channels?
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Posts and attachments in archived channels are affected by the data retention job. If a post exceeds the age configured for the data retention job it will be deleted from the database.
+
+How long does it take to run a deletion query and does it affect server performance?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Data retention runs the actual deletion query in batches, deleting data in blocks of 1 000 records per query. This is so the database won’t be locked up for extended periods of time with long-running queries. Keeping to this limit keeps the query down to a few milliseconds' execution time on the database itself.
+
+Each batch of data is deleted based on indexes, so the queries are generally quite quick to execute on such small batches meaning the server will stay fully responsive while the process is running.
```
Comment (justinegeffen) [inline source/administration/data-retention.rst:117]:
```suggestion
Each batch of data is deleted based on indexes making the queries quick to execute on small batches. This helps the server remain fully responsive while the process is running.
```

File: source/administration/data-retention.rst (hunk 2 above)
Comment (justinegeffen) [reply source/administration/data-retention.rst:117]: I really like it, thanks. I've added a new suggestion with your change. :)

File: source/administration/data-retention.rst [hunk 7]
```diff
@@ -108,3 +108,19 @@ Does the data retention job include archived channels?
...
+
+How do I know whether the data retention job is running/scheduled?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The job scheduler runs the data retention job based on the time specified in the configuration settings. At this time a ``DEBUG``-level log line is printed: ``Scheduling data retention job``.
+
+When a job server then picks up that scheduled job in order to execute it there will be a debug log as follows: ``Worker EnterpriseDataRetention: Received a new candidate job``.
+
+When the job is complete, an ``INFO``-level log line ``Worker EnterpriseDataRetention: Job is complete`` is generated.
```
Comment (justinegeffen) [inline source/administration/data-retention.rst:126]:
```suggestion
When the job is complete, an ``INFO``-level log line is generated: ``Worker EnterpriseDataRetention: Job is complete``. 
```

**Comment by fakela** [comment]
@justinegeffen everything looks good now 💯

**Comment by justinegeffen** [comment]
> @justinegeffen everything looks good now 💯

Yay, teamwork! Glad it works. :) Just waiting on PM review and then I think we can merge. :)

**Pull request merged**